packages/xo-server/src/xo-mixins/backups-ng/index.js

```diff
@@ -1613,6 +1613,17 @@ export default class BackupNg {
                     _ => _.mode === 'delta' && _.scheduleId === scheduleId
                   )
                 ): any)
+
+                // FIXME: implement optimized multiple VHDs merging with
+                // synthetic delta
+                //
+                // For the time being, limit the number of deleted backups to 2
+                // because it can time a very long time and can lead to
+                // interupted backup with broken VHD chain.
+                if (oldBackups.length > 2) {
+                  oldBackups.length = 2
```

Comment:
Can you explain this a bit more? So if the user lowers the retention by a lot then, on the next run, XO will have to merge many VHDs but we decide to force it to only merge the 2 oldest ones. And then the delta chain may still be long but will be shortened slowly but surely on each run by merging 2 backups (and adding one) every time until it reaches the requested retention.

Correct?

Reply:
Correct, do you want me to update the comment?

Reply:
Maybe just to say that the other old backups will be merged later by the same mechanism. You can guess it but if you see it written, you don't have to assume it.